redshift_connector/core.py

```diff
@@ -2003,7 +2003,22 @@ def handle_messages(self: "Connection", cursor: Cursor) -> None:
         code = self.error = None
 
         while code != READY_FOR_QUERY:
-            code, data_len = ci_unpack(self._read(5))
+            buffer = self._read(5)
+
+            if len(buffer) == 0:
+                if self._usock.timeout is not None:
+                    raise InterfaceError(
+                        "BrokenPipe: server socket closed. We noticed a timeout is set for this connection. Consider "
+                        "raising the timeout or defaulting timeout to none."
+                    )
+                else:
+                    raise InterfaceError(
+                        "BrokenPipe: server socket closed. Please check that client side networking configurations such "
+                        "as Proxies, firewalls, VPN, etc. are not affecting your network connection."
+                    )
+
+            code, data_len = ci_unpack(buffer)
+
             self.message_types[code](self._read(data_len - 4), cursor)
```

Comment:
do we need to perform the same empty buffer check on subsequent calls to `._read`?